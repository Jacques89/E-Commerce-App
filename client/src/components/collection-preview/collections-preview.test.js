import React from 'react'
import { shallow } from 'enzyme'
import CollectionPreviewDiv from './collection-preview.component'

describe('CollectionPreview Component Tests', () => {
    let wrapper
    let mockHistory
    let mockMatch
    let mockRouteName = 'vinyl'
    let mockItem

    beforeEach(() => {
        mockHistory = {
            push: jest.fn()
        }

        mockMatch = {
            path: '/shop'
        }

        mockItem = {
            id: 1,
            imageUrl: 'https://i.ibb.co/0s3pdnc/adidas-nmd.png',
            price: 220,
            name: 'Adidas NMD'
        }

        const mockProps = {
            title: 'vinyl',
            items: [mockItem],
            history: mockHistory,
            match: mockMatch,
            routeName: 'vinyl'
        }

        wrapper = shallow(<CollectionPreviewDiv.WrappedComponent {...mockProps} />)
    })

    it('should render and match snapshot', () => {
        expect(wrapper).toMatchSnapshot()
    })

    it('should fire history.push onClick of title', () => {
        wrapper.find('TitleStyles').simulate('click')
        expect(mockHistory.push).toHaveBeenCalledWith(`${mockMatch.path}/${mockRouteName}`)
    })
})
