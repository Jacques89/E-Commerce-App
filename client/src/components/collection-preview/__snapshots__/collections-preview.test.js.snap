// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollectionPreview Component Tests should render and match snapshot 1`] = `ShallowWrapper {}`;
